Design.md

```diff
@@ -229,6 +231,20 @@ This API will enable or disable user on Vault.
 ### Create S3 Credential
 This API creates S3 credential for the user.
 1. `generate-access-key` api will be called using assumed role credentials.
+1. Encrypt the `secret-key` using AES algorithm and store the encrypted `secret-key` on Redis Cache in the hash named as `OSIS` with the hash-key formatted as `<Username>__<AccessKeyID>`.
+1. Log the `access-key-id` _(Log location and tools for logging are yet to be decided)_
+
+**Important Notes:**
+* Redis cache shall always be rolling deployed.
+* I total Redis Sentinel fails or crashes,  vCloud Director OSE service needs to be restarted with the following command._
+    ```shell
+    $ ose service restart
+    ```
+  * All the access keys created by OSIS before the `ose-restart-time` will be dead access keys on Vault DB which OSE/OSIS will never be able to use and can only be deleted manually from the Vault. 
+  * A log location for all the dead access key IDs created by OSIS, and a script to run manually will be provided.
```

Comment:
What does the script do? Why would I want to run it?

Reply:
+1

Reply:
```suggestion
  * A log location for all the dead access key IDs created by OSIS, and a script to run manually, to delete these dead keys from Vault DB, will be provided.
```